### PR TITLE
Extend retries for file-ops on windows

### PR DESCRIPTION
**Pull request by sigurdm**
Attempt to mitigate/fix https://github.com/dart-lang/pub/issues/3334.

### CODE REVIEW

**Review by jonasfj — APPROVED** [review]

File: lib/src/io.dart [hunk 1]
```diff
@@ -390,18 +390,20 @@ void _attempt(String description, void Function() operation,
...
     } on FileSystemException catch (error) {
       var reason = getErrorReason(error);
       if (reason == null) rethrow;
 
-      if (i < 2) {
+      if (i < maxRetries - 1) {
+        final delayMilliseconds = 50 * (i + 1);
         log.io('Pub failed to $description because $reason. '
-            'Retrying in 50ms.');
-        sleep(Duration(milliseconds: 50));
```
Comment (jonasfj) [inline lib/src/io.dart:404]: this is low cost, we can just do it every 50ms and give it more retries...

**Review by jonasfj — APPROVED** [review]

**Comment by sigurdm** [comment]
@timsneath do you have a chance to try this - and see if it indeed helps with: https://github.com/dart-lang/pub/issues/3334 ?

File: lib/src/io.dart (hunk 1 above)
Comment (sigurdm) [reply lib/src/io.dart:404]: Done

**Comment by sigurdm** [comment]
The failures are from changes in the sdk used for testing and unrelated to this PR.

@alexmarkov could the changes in stack-trace be related to https://github.com/dart-lang/sdk/commit/af4da780be986e9aad22ab18395f1548ac1d1937 ? 

https://github.com/dart-lang/pub/runs/6813277082?check_suite_focus=true

**Pull request merged**